code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -57,6 +57,20 @@ var/global/list/datum/stack_recipe/metal_recipes = list ( \
 	flags = CONDUCT
 	origin_tech = "materials=1"
 
+/obj/item/stack/sheet/metal/attackby(obj/item/weapon/W, mob/living/user)
+	..()
+	if(istype(W, /obj/item/weapon/paper/talisman/construction))
```

Comment:
Actually, in afterattack(); just do:
````
/obj/item/weapon/paper/talisman/construction/afterattack(obj/item/stack/sheet/metal/target, mob/user, proximity_flag, click_parameters)
	if(proximity_flag && istype(target) && iscultist(user))
		{what you have here, except with the new /obj/structure/constructshell(get_turf(src)) line using target instead of src}
````